### ISSUE TITLE
No response from test sketchs for  DS28CM00

### ISSUE THREAD

**Issue by dlowerymcse** [RobTillaart/DS28CM00#2]
Hello,
I am new to GIT hub and Arduino.  I do not know the protocol for asking questions. 

Through a google search, your library is the only instance I have found where anyone is attempting to use the DS28CM00 chip with an Arduino.  I have installed your latest library files, and attempting to run them on a Nano using the latest IDE ver. 1.8.13.  I get no response from the test or perf sketch.  They do not seem to even get to the setup void.  I tried across multiple nano boards and on a windows and linux machine. 

I was able to get data reads from the chip using a basic I2C loop(see below) and the data is valid so I know the Nano and chip are functioning correctly. My simple loop will not reliably start at the 00 address.  

`
#include <Wire.h>

void setup() {

  Serial.begin(9600);
  delay(100);
  Wire.begin(); 
}

void loop() {
  Wire.requestFrom(0x50, 8);
  for (int i = 0; i < 8; i++) {
     Serial.println(Wire.read(), HEX);
  }
  Serial.println("");
  delay(1000);
}`

The result should be 7029826601000089
Examples of what I get out are as follows:

70
29
82
66
1
0
0
89

1
70
29
82
66
1
0
0

89
1
70
29
82
66
1
0

**Comment by RobTillaart** [comment]
Goodmornig,
This is the right way to ask questions on GitHub. You mention the problem, how to recreate it and tell about your hardware and code used.

Tip: read about the markdown language, that allows you to do bold italic and things like bulletl ists and most important code blocks. 

I will dive into the problem later today.

**Comment by RobTillaart** [comment]
### CODE
to get better output try this

```cpp
#include <Wire.h>

void setup()
{
  Serial.begin(9600);
  Serial.println(__FILE__);  // show sketch name
  delay(100);
  Wire.begin();
}

void loop() 
{
  n = Wire.requestFrom(0x50, 8);
  if (n == 8)
  {
    for (int i = 0; i < 8; i++) 
    {
      int x = Wire.read();
      if (x < 16) Serial.print("0");  // print leading zero
      Serial.print(x, HEX);
      Serial.print(" ");
    }
  }
  Serial.println("");
  delay(1000);
}
```

### OUTPUT

The output is ~90% OK but somehow an additional value 1 sneaks in, 
- to be investigated, need to see if I have all components to set up a test .

**Comment by RobTillaart** [comment]
No chip nearby so hard to test over here.
This sketch adds the explicit setting of the UID register.

```cpp
#include <Wire.h>

void setup()
{
  Serial.begin(9600);
  Serial.println(__FILE__);  // show sketch name
  delay(100);
  Wire.begin();
}

void loop() 
{
  Wire.beginTransmission(0x50);  // explicitly set UID register 
  Wire.write(0x00);
  Wire.endTransmission();

  n = Wire.requestFrom(0x50, 8);
  if (n == 8)
  {
    for (int i = 0; i < 8; i++) 
    {
      int x = Wire.read();
      if (x < 16) Serial.print("0");  // print leading zero
      Serial.print(x, HEX);
      Serial.print(" ");
    }
  }
  Serial.println("");
  delay(1000);
}
```

**Comment by RobTillaart** [comment]
An alternative could be to always just read 9 bits but only show 8
```cpp
#include <Wire.h>

uint8_t x[9];  // array to hold values

void setup()
{
  Serial.begin(9600);
  Serial.println(__FILE__);  // show sketch name
  delay(100);
  Wire.begin();
}

void loop() 
{
  int n = Wire.requestFrom(0x50, 9);
  if (n == 9)
  {
    for (int i = 0; i < 9; i++)  x[i] = Wire.read();
    for (int i = 0; i < 8; i++) 
    {
      if (x[i] < 16) Serial.print("0");  // print leading zero
      Serial.print(x[i], HEX);
      Serial.print(" ");
    }
  }
  Serial.println("");
  delay(1000);
}
```

**Comment by dlowerymcse** [comment]
Thank you Robert. 

I am amazed at your quick response. The code you have supplied works beautifully.   I had programmed this on an RPi in BASH and Python without issue but, was really struggling with it on the Arduino.   As these chips are so cheap and easily available in the US I was surprised to not see them used more.  I found several instances where others had referenced your library but, not a single example of usage.  

Again, thank you very much!  I can now move forward with my project and certainly can learn from details you have included in the example code.
